contrib/sonatypecentral/src/mill/contrib/sonatypecentral/SonatypeCentralPublishModule.scala

```diff
@@ -0,0 +1,143 @@
+package mill.contrib.sonatypecentral
+
+import com.lumidion.sonatype.central.client.core.{PublishingType, SonatypeCredentials}
+import mill._
+import scalalib._
+import define.{ExternalModule, Task}
+import mill.api.Result
+import mill.contrib.sonatypecentral.SonatypeCentralPublishModule.{
+  defaultAwaitTimeout,
+  defaultConnectTimeout,
+  defaultCredentials,
+  defaultReadTimeout,
+  getPublishingTypeFromReleaseFlag,
+  getSonatypeCredentials
+}
+import mill.scalalib.PublishModule.{defaultStringGpgArgs, getFinalGpgArgs}
+import mill.scalalib.publish.Artifact
+import mill.scalalib.publish.SonatypeHelpers.{
+  PASSWORD_ENV_VARIABLE_NAME,
+  USERNAME_ENV_VARIABLE_NAME
+}
+
+trait SonatypeCentralPublishModule extends PublishModule {
+  def gpgArgs: T[String] = T { defaultStringGpgArgs }
+
+  def connectTimeout: T[Int] = T { defaultConnectTimeout }
+
+  def readTimeout: T[Int] = T { defaultReadTimeout }
+
+  def awaitTimeout: T[Int] = T { defaultAwaitTimeout }
+
+  def shouldRelease: T[Boolean] = T { true }
```

Comment:
I think it would by good to prefix these target names with something unique to your module, e.g. `sonatype` or `sonatypeCentral`. This makes reading larger build files much easier.

Reply:
Prefixed all with `sonatypeCentral`.